pkg/cmd/codespace/ssh.go

```diff
@@ -298,6 +338,121 @@ func checkAndUpdateOldKeyPair(sshContext ssh.Context) *ssh.KeyPair {
 	return nil
 }
 
+// hasUploadedValidCustomKey checks which private keys are used in ssh configration and
+// compares them to uploaded public keys to see if there is a match
+func hasUploadedPublicKeyForConfig(
+	ctx context.Context,
+	sshContext ssh.Context,
+	apiClient apiClient,
+	customConfigFile string,
+	customHost string,
+) (bool, error) {
+	configuredPrivateKeyPaths, err := getConfiguredPrivateKeys(ctx, customConfigFile, customHost)
+	if err != nil {
+		return false, fmt.Errorf("getting local ssh keys: %w", err)
+	}
+
+	configuredPublicKeys := []string{}
```

Comment:
nit (here and others): https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices

It's more idiomatic to declare `var configuredPublicKeys []string` than an initialized slice to zero length.

Reply:
Updated